src/FakeItEasy.Analyzer/UnusedReturnValueAnalyzer.cs

```diff
@@ -67,11 +58,31 @@ private static IMethodSymbol GetCalledMethodSymbol(InvocationExpressionSyntax ca
             return context.SemanticModel.GetSymbolInfo(memberAccess?.Name).Symbol as IMethodSymbol;
         }
 
-        private static string GetDiagnosticName(IMethodSymbol methodSymbol)
+        private static DiagnosticDescriptor GetDiagnostic(IMethodSymbol methodSymbol)
         {
-            return methodSymbol.GetAttributes()
-                .FirstOrDefault(a => a.AttributeClass.GetFullName() == "FakeItEasy.Analysis.MustUseReturnValueAttribute")
-                ?.ConstructorArguments.FirstOrDefault().Value as string;
+            string typeName = methodSymbol.ContainingType.GetFullName();
+            string fullName = typeName + "." + methodSymbol.GetDecoratedName();
+            return DiagnosticsMap.GetValueOrDefault(fullName);
+        }

```

Comment:
I just introduced a variable to make it clear that we're passing the full name of the method